datasync/chaindatafetcher/repository.go

```diff
@@ -30,3 +35,19 @@ type Repository interface {
 
 	SetComponent(component interface{})
 }
+
+type retry func(blockchain.ChainEvent) error
+
+func retryFunc(f retry) retry {
+	return func(event blockchain.ChainEvent) error {
+		var err error
+		for {
+			if err = f(event); err == nil {
+				break
+			}
+			logger.Warn("retrying...", "blockNumber", event.Block.NumberU64())
```

Comment:
How about leaving retry count in the log?

Reply:
Added on 9659b14b55b6cece1acef7175860f321a84daf4b